src/cs/Ssh.Tcp/Services/PortForwardingService.cs

```diff
@@ -30,12 +30,38 @@ public class PortForwardingService : SshService
 	internal const string ReversePortForwardChannelType = "direct-tcpip";
 
 	private bool disposed;
+
+	/// <summary>
+	/// Maps from local IP address and FORWARDED port number to the object that manages
+	/// listening for incoming connections on that port and forwarding them through the session.
+	/// </summary>
+	/// <remarks>
+	/// Note the actual local source port number used may be different from the forwarded port
+	/// number if the local TCP listener factory chose a different port. The forwarded port number
+	/// is used to identify the port in any messages exchanged between client and server.
+	/// </remarks>
 	private readonly IDictionary<IPEndPoint, LocalPortForwarder> localForwarders =
 		new Dictionary<IPEndPoint, LocalPortForwarder>();
-	private readonly IDictionary<IPEndPoint, RemotePortConnector> remoteConnectors =
-		new Dictionary<IPEndPoint, RemotePortConnector>();
-	private readonly ICollection<ChannelForwarder> channelForwarders =
-		new List<ChannelForwarder>();
+
+	/// <summary>
+	/// Maps from FORWARDED port numbers to the object that manages relaying forwarded connections
+	/// from the session to a local port.
+	/// </summary>
+	/// <remarks>
+	/// Note the actual local destination port number used may be different from the forwarded port
+	/// number. The forwarded port number is used to identify the port in any messages exchanged
+	/// between client and server.
+	/// </remarks>
+	private readonly IDictionary<int, RemotePortConnector> remoteConnectors =
```

Comment:
Does the key need to be `IPEndPoint` as well, if port forwarding can forward from same port of different interfaces?

Reply:
That is not supported because the protocol does not have a way to distinguish those when opening channels.